src/basic/cgroup-util.c

```diff
@@ -90,6 +90,24 @@ int cg_read_pid(FILE *f, pid_t *ret) {
         return 1;
 }
 
+int cg_read_pidref(FILE *f, PidRef *ret) {
+        int r;
+
+        for (;;) {
+                pid_t pid;
+
+                r = cg_read_pid(f, &pid);
+                if (r <= 0) /* failure or EOF? */
+                        return r;
+
+                r = pidref_set_pid(ret, pid);
+                if (r != -ESRCH)
+                        return r;
```

Comment:
If `0` is returned in the above, then maybe it is better to return `1` on success here. Otherwise, we need to call `pidfd_is_set()` to check if the function return valid pidref on success.